pyega3/data_file.py

```diff
@@ -0,0 +1,278 @@
+            path += '?' + urllib.parse.urlencode(options)
+
+        file_name += f'-from-{str(start_pos)}-len-{str(length)}.slice'
+        self.temporary_files.add(file_name)
+
+        existing_size = os.stat(file_name).st_size if os.path.exists(file_name) else 0
+        if existing_size > length:
+            os.remove(file_name)
+        if pbar:
+            pbar.update(existing_size)
+
+        if existing_size == length:
+            return file_name
+
+        with self.data_client.get_stream(path,
+                                         {'Range': f'bytes={start_pos + existing_size}-{start_pos + length - 1}'}) as r:
+            with open(file_name, 'ba') as file_out:
+                for chunk in r.iter_content(DOWNLOAD_FILE_SLICE_CHUNK_SIZE):
+                    file_out.write(chunk)
+                    if pbar:
+                        pbar.update(len(chunk))
+
+        total_received = os.path.getsize(file_name)
+        if total_received != length:
+            raise Exception(f"Slice error: received={total_received}, requested={length}, file='{file_name}'")
+
+        return file_name
+
+    @staticmethod
+    def is_genomic_range(genomic_range_args):
+        if not genomic_range_args:
+            return False
+        return genomic_range_args[0] is not None or genomic_range_args[1] is not None
+
+    def generate_output_filename(self, folder, genomic_range_args):
+        file_name = self.display_name
+        ext_to_remove = ".cip"
+        if file_name.endswith(ext_to_remove):
+            file_name = file_name[:-len(ext_to_remove)]
+        name, ext = os.path.splitext(os.path.basename(file_name))
+
+        genomic_range = ''
+        if DataFile.is_genomic_range(genomic_range_args):
+            genomic_range = "_genomic_range_" + (genomic_range_args[0] or genomic_range_args[1])
+            genomic_range += '_' + (str(genomic_range_args[2]) or '0')
+            genomic_range += '_' + (str(genomic_range_args[3]) or '')
+            format_ext = '.' + (genomic_range_args[4] or '').strip().lower()
+            if format_ext != ext and len(format_ext) > 1:
+                ext += format_ext
+
+        ret_val = os.path.join(folder, self.id, name + genomic_range + ext)
+        logging.debug(f"Output file:'{ret_val}'")
+        return ret_val
+
+    @staticmethod
+    def print_local_file_info_genomic_range(prefix_str, file, gr_args):
+        logging.info(
+            f"{prefix_str}'{os.path.abspath(file)}'({os.path.getsize(file)} bytes, referenceName={gr_args[0]},"
+            f" referenceMD5={gr_args[1]}, start={gr_args[2]}, end={gr_args[3]}, format={gr_args[4]})"
+        )
+
+    def download_file_retry(self, num_connections, output_file, genomic_range_args, max_retries, retry_wait, key=None):
+        if self.name.endswith(".gpg"):
+            logging.info(
+                "GPG files are not supported, please use the Java client"
+                " - https://ega-archive.org/download/using-ega-download-client")
```

Comment:
Do we still have a Java client?
Anyway, this URL redirects to the python client. I think we should create a ticket (or at least a TODO item) to remove this misleading message.

Reply:
```suggestion
                " Please email EGA Helpdesk at helpdesk@ega-archive.org")
```

Reply:
@bela-juhasz I'm going to change the sentence for the one that @malloryfreeberg suggested in her comment